packages/nextjs/src/common/helpers.ts

```diff
@@ -107,9 +112,45 @@ export function addToCookies(newCookies: string[], res: ServerResponse) {
   }
   res.setHeader('set-cookie', [...existingSetCookie, ...newCookies]);
 }
+const createEmptySingleCookie = (cookieName: string, isSecured: boolean, cookieDomain: string) =>
+  createCookie({ cookieName, session: '', expires: new Date(), isSecured, cookieDomain });
+
+const createEmptyCookies = (cookieName: string, isSecured: boolean, cookieDomain: string, numberOfCookies: number) => {
+  if (!numberOfCookies || numberOfCookies === 1) {
+    return createEmptySingleCookie(cookieName, isSecured, cookieDomain);
+  }
+  const allEmptyCookies = [];
+  for (let i = 0; i < numberOfCookies; i++) {
+    allEmptyCookies.push(...createEmptySingleCookie(`${cookieName}-${i}`, isSecured, cookieDomain));
+  }
+  return allEmptyCookies;
+};
+
+export const getNumberOfCookiesFromRequest = (req?: RequestType): number => {
+  let numberOfCookies = 1;
+  if (!req) {
+    return numberOfCookies;
+  }
+  const cookieStr = 'credentials' in req ? req.headers.get('cookie') || '' : req.headers.cookie || '';
+  const allCookies = cookieStr && cookie.parse(cookieStr);
+  if (!allCookies || allCookies[FronteggConfig.cookieName]) {
+    return numberOfCookies;
+  }
+  while (allCookies[`${FronteggConfig.cookieName}-${numberOfCookies}`]) {
+    numberOfCookies++;
+  }
+  return numberOfCookies - 1;
```

Comment:
consider change the cookie structure to be zero-based (start with 0)

Reply:
will keep like this for now